core/src/main/java/org/apache/metamodel/query/ConcatFunction.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.query;
+
+import org.apache.metamodel.data.Row;
+import org.apache.metamodel.schema.Column;
+import org.apache.metamodel.schema.ColumnType;
+import org.apache.metamodel.util.CollectionUtils;
+import org.apache.metamodel.util.Predicate;
+
+import java.util.List;
+
+/**
+ * Represents a function that concatenates values.
+ */
+public final class ConcatFunction extends DefaultScalarFunction {
+
+    @Override
+    public Object evaluate(Row row, Object[] parameters, SelectItem operandItem) {
+        SelectItem[] selectItems = row.getSelectItems();
+        if (parameters.length == 0) {
+            throw new IllegalArgumentException("Expecting some parameters to CONCAT function");
+        }
+        StringBuilder strBuilder = new StringBuilder();
+        for (Object parameter: parameters) {
+            String parameterAsString = parameter.toString();
```

Comment:
Wouldn't it make a lot of things easier if we decided to treat the parameters as SelectItems instead of Strings? What I mean is - in this part of the code (and in ScalarFunctionRow) the the changes made are quite involved because the parameters are of type String. But what if we instead changed it so that the parameters are simply SelectItems. I guess that is also what they _are_ from a (SQL) language perspective: ```CONCAT(foo, bar)``` would indicate that "foo" and "bar" are select items, whereas ```CONCAT(foo, " ", bar)``` would be a select item, a string and then a select item. I think we could greatly simplify the code if we work with this model instead.